libopenage/pathfinding/cost_field.cpp

```diff
@@ -56,6 +56,27 @@ void CostField::set_costs(std::vector<cost_t> &&cells, const time::time_t &valid
 	this->valid_until = valid_until;
 }
 
+bool CostField::stamp(size_t idx, cost_t cost, const time::time_t &stamped_at) {
+	if (this->cost_stamps[idx].has_value()) return false;
+
+	cost_t original_cost = this->get_cost(idx);
+	this->cost_stamps[idx]->original_cost = original_cost;
+	this->cost_stamps[idx]->stamp_time = stamped_at;
+
+	this->set_cost(idx, cost, stamped_at);
+	return true;
+}
+
+bool CostField::unstamp(size_t idx, const time::time_t &unstamped_at) {
+	if (!this->cost_stamps[idx].has_value() || unstamped_at < this->cost_stamps[idx]->stamp_time) return false;
```

Comment:
if conditions should always be multiline and wrapped in curly brackets, according to our code style conventions. Also, you should use `or`, `not`, and `and` instead of `||`, `!`, and `&&` for readability in the condition.

```suggestion
	if (not this->cost_stamps[idx].has_value() or unstamped_at < this->cost_stamps[idx]->stamp_time) {
	    return false;
	}
```